l5kit/l5kit/visualization/visualiser/visualiser.py

```diff
@@ -0,0 +1,133 @@
+from typing import Dict, List, Optional
+
+import bokeh.io
+import bokeh.plotting
+import numpy as np
+from bokeh.layouts import column
+from bokeh.models import CustomJS, HoverTool, Slider
+from bokeh.plotting import ColumnDataSource, output_notebook, save, show
+from bokeh.resources import Resources
+
+from l5kit.visualization.visualiser.common import (AgentVisualisation, CWVisualisation, EgoVisualisation,
+                                                   FrameVisualisation, LaneVisualisation, TrajectoryVisualisation)
+
+
+def visualise(scene_index: int, frames: List[FrameVisualisation], do_show: bool = False,
+              do_notebook: bool = False, save_path: Optional[str] = None,
+              resources: str = "inline") -> None:
+    """Visualise a scene using Bokeh.
+
+    :param scene_index: the index of the scene, used only as the title
+    :param frames: a list of FrameVisualisation objects (one per frame of the scene)
+    :param do_show: if to call the show method
+    :param do_notebook: if to call output_notebook
+    :param save_path: optional html destination file
+    :param resources: how to load js (inline will embed it, cdn will download it)
+    """
+
+    assert not (save_path is not None and do_notebook), "can't save and show in notebook"
+
+    if do_notebook:
+        output_notebook()
+
+    agent_hover = HoverTool(
+        mode="mouse",
+        names=["agents"],
+        tooltips=[
+            ("Type", "@type"),
+            ("Probability", "@prob{0.00}%"),
+            ("Track id", "@track_id"),
+        ],
+    )
+    out: List[Dict[str, ColumnDataSource]] = []
+
+    trajectories_labels = np.unique([traj.legend_label for frame in frames for traj in frame.trajectories])
+
+    for frame_idx in range(len(frames)):
+        ego = frames[frame_idx].ego
+
+        ego_dict = {field_name: [field] for field, field_name in zip(ego, EgoVisualisation._fields)}
+
+        agents_dict = {k: [] for k in AgentVisualisation._fields}
+        for agent in frames[frame_idx].agents:
+            for k, v in zip(AgentVisualisation._fields, agent):
```

Comment:
```suggestion
            for k, v in agent._asdict().items():
```
Same for the others below. I would consider adding a function to do that since we're repeating the code below.

Reply:
function added